lib/Elevate/Blockers.pm

```diff
@@ -112,6 +113,50 @@ sub check ( $self, %opts ) {    # do_check - main  entry point
     return $has_blockers;
 }
 
+sub bail_if_check_not_run_or_blockers_found () {
+
+    if ( !check_run_recently() ) {
+        FATAL("You must first run script with --check to check if your server is ready to upgrade");
+        exit 1;
+    }
+
+    my $blocker_ar = load_past_blockers();
+
+    # Pass if no blockers in blocker file
+    return unless scalar @$blocker_ar;
+
+    say "The following blockers must still be resolved:";
+    foreach my $bl (@$blocker_ar) {
+        say '';
+        say $bl->{id} . ':';
+        say $bl->{msg};
+    }
+
+    FATAL("You must fix all blockers and re-run with --check before performing the upgrade.");
+    exit 1;
+}
+
+sub check_run_recently () {
+    return 0 if ( !-f ELEVATE_BLOCKER_FILE );
+
+    return ( ( stat(ELEVATE_BLOCKER_FILE) )[9] + ELEVATE_BLOCKER_FILE_EXPIRE_TIME > time() ) ? 1 : 0;
```

Comment:
You should be able to test this with Test::MockFile. From my cPanel blocker experience, we often got this logic wrong :(

Reply:
It didn't look like Elevate::Blockers had a unit test; I guess I could start one & test the functions I added in it.